engine.c

```diff
@@ -519,20 +475,10 @@ assign_ptrack_map_size(int newval, void *extra)
 	elog(DEBUG1, "assign_ptrack_map_size: MyProc %d newval %d ptrack_map_size " UINT64_FORMAT,
 		 MyProcPid, newval, ptrack_map_size);
 
-	/*
-	 * XXX: for some reason assign_ptrack_map_size is called twice during the
-	 * postmaster boot!  First, it is always called with bootValue, so we use
-	 * -1 as default value and no-op here.  Next, it is called with the actual
-	 * value from config.  That way, we use 0 as an option for user to turn
-	 * off ptrack and clean up all files.
-	 */
-	if (newval == -1)
-		return;
```

Comment:
Is it not a problem anymore? I see you use 0 as a boot value and clean up files when ptrack_map_size == 0

Reply:
We now do nothing (with files and memory) inside assign_ptrack_map_size() and use the value of ptrack_map_size after the re-assignment has passed.

Well, at least that's how I understand it.